main.tf

```diff
@@ -100,19 +105,21 @@ data "aws_iam_policy_document" "ssm_activation" {
 }
 
 resource "aws_iam_role" "ssm_activation" {
+  count              = var.ssm_activation_role_id != null ? 0 : 1
```

Comment:
Since you're working with locals already, you can make this a local as well

Reply:
done